2008-13-08  Samuel Thibault  <samuel.thibault@ens-lyon.org>

	* hurd/hurd/signal.h (HURD_MSGPORT_RPC): Add parenthesis around
	assignment used as truth value.

Index: glibc/hurd/hurd/signal.h
===================================================================
RCS file: /cvs/glibc/libc/hurd/hurd/signal.h,v
retrieving revision 1.49
diff -u -p -r1.49 signal.h
--- glibc/hurd/hurd/signal.h	19 Jul 2007 17:20:39 -0000	1.49
+++ glibc/hurd/hurd/signal.h	12 Aug 2008 23:16:59 -0000
@@ -342,10 +342,10 @@ extern mach_msg_timeout_t _hurd_interrup
     do									      \
       {									      \
 	/* Get the message port.  */					      \
-	if (__err = (fetch_msgport_expr))				      \
+	if ((__err = (fetch_msgport_expr)))				      \
 	  break;							      \
 	/* Get the reference port.  */					      \
-	if (__err = (fetch_refport_expr))				      \
+	if ((__err = (fetch_refport_expr)))				      \
 	  {								      \
 	    /* Couldn't get it; deallocate MSGPORT and fail.  */	      \
 	    __mach_port_deallocate (__mach_task_self (), msgport);	      \
